plugins/engine-configuration/impl/src/test/java/org/pentaho/di/engine/configuration/impl/RunConfigurationProviderFactoryManagerImplTest.java

```diff
@@ -0,0 +1,66 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.engine.configuration.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.pentaho.di.core.service.PluginServiceLoader;
+import org.pentaho.di.engine.configuration.api.CheckedMetaStoreSupplier;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProvider;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProviderFactory;
+import org.pentaho.di.engine.configuration.impl.pentaho.DefaultRunConfigurationProviderFactory;
+import org.pentaho.metastore.stores.memory.MemoryMetaStore;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class RunConfigurationProviderFactoryManagerImplTest {
+
+  @Test
+  public void testDefaultProvidedMetastore() {
+    MemoryMetaStore memoryMetaStore = new MemoryMetaStore();
+    CheckedMetaStoreSupplier metastoreSupplier = () -> memoryMetaStore;
```

Comment:
Not a blocker, but think MemoryMetaStorejava is going away or not really used. But for a unit test, you should try and not rely on a class not under test to provide functionality.

Reply:
I will be updating this to [MetastoreConst as Alan pointed out](https://github.com/pentaho/pentaho-kettle/pull/9966#discussion_r2056705325)

I will update this comment with the link to the change...

Reply:
I will centralize related discussion [in this thread](https://github.com/pentaho/pentaho-kettle/pull/9966#discussion_r2056812071)